### PR TITLE
Added a CircleCI job to validate contributors

### PR DESCRIPTION
**Pull request by simpleigh**
## Why?

All contributors should sign our Contributor License Agreement to confirm they agree to donate their code to the HawkEar project. We'd prefer to enforce this automatically.

## What?

* Add `CONTRIBUTORS.md` file. People "sign" the agreement by adding their GitHub username to this file.
* Add CircleCI job to check a PR's author is in the contributors file.

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Contributors
+
+Adding your GitHub username to this list confirms that you agree to the terms of
+the Contributor License Agreement.
+
+simpleigh
+drichards2
```
Comment (simpleigh) [inline CONTRIBUTORS.md:7]: @drichards2 please could we keep this file sorted? One day there might be many more entries. I'd just do this, but am quite keen that `git blame` has your name against the line!

Comment (simpleigh) [reply CONTRIBUTORS.md:7]: (I'd just do this, but am quite keen that `git blame` has your name against the line!)

**Review by ianthehawk — APPROVED** [review]

File: CONTRIBUTORS.md (hunk 1 above)
Comment (drichards2) [reply CONTRIBUTORS.md:7]: Mmmm ... what's the advantage of having it sorted? I can't imagine there's any use case for the file other than searching for a known token and even if there were thousands of contributors, presumably we'd never rely on it being sorted for efficiency reasons (?)

Comment (simpleigh) [reply CONTRIBUTORS.md:7]: No technical reason, just to make it easier for humans. Admittedly applying this criterion is making it harder for one human (you!). Happy either way.

**Review by PJopp — APPROVED** [review]

**Review by PJopp — APPROVED** [review]

**Pull request merged**